include/hlslib/common/OpenCL.h

```diff
@@ -499,6 +502,97 @@ class Buffer {
   Buffer(Context &context, size_t nElements)
       : Buffer(context, MemoryBank::unspecified, nElements) {}
 
+#ifdef HLSLIB_XILINX
+  /// Allocate DDR or HBM but don't perform any transfers.
+  Buffer(Context &context, StorageType storageType, uint8_t bankIndex, size_t nElements)
+      : context_(&context), nElements_(nElements) {
+#ifndef HLSLIB_SIMULATE_OPENCL
+
+    cl_mem_flags flags;
+    switch (access) {
+      case Access::read:
+        flags = CL_MEM_READ_ONLY;
+        break;
+      case Access::write:
+        flags = CL_MEM_WRITE_ONLY;
+        break;
+      case Access::readWrite:
+        flags = CL_MEM_READ_WRITE;
+        break;
+    }
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)! Instead of repeating all of this, make a single function that does the allocation. The other function should translate the arguments into the canonical format. We can use the new format as the canonical format, so the `MemoryBank0` should be translated into `DDR, 0` and call the other function.